hardware_interface/include/hardware_interface/components/joint.hpp

```diff
@@ -84,7 +84,7 @@ class Joint
   HARDWARE_INTERFACE_EXPORT
   return_type get_command(
     std::vector<double> & command,
-    const std::vector<std::string> & interfaces) const;
+    const std::vector<InterfaceInfo> & interfaces) const;
```

Comment:
For the API simplicity, we should probably stay with `std::string` as parameters